src/trip_kinematics/URDFParser.py

```diff
@@ -0,0 +1,301 @@
+        axis = joint.find('axis')
+        # Default to (0, 0, 1) if the axis is unspecified. Note that there does not seem to be
+        # a standard default value for the axis. Maybe better to remove the default to prevent
+        # potential confusion?
+        if axis is None:
+            axis = '0 0 1'
+        else:
+            axis = axis.get('xyz')
+        axis = np.array(list(map(float, axis.split(' '))))
+        axis /= np.linalg.norm(axis)
+
+    # For each joint, define up to five transformations
+    # Transformation 1 (tra): defined by the translation of the origin
+    tra = [name + '_tra', {'tx': origin_xyz[0], 'ty': origin_xyz[1], 'tz': origin_xyz[2]}, []]
+
+    # Transformation 2 (rot): defined by the rotation of the origin
+    rot_quat = ScipyRotation.from_euler('xyz', [*origin_rpy], degrees=False).as_quat()
+    rot = [name + '_rot',
+           {'qw': rot_quat[3],
+            'qx': rot_quat[0],
+            'qy': rot_quat[1],
+            'qz': rot_quat[2]},
+           [],
+           ]
+
+    joint_transformations.extend([tra, rot])
+
+    if axis is not None and not np.array_equal(axis, np.aray([0, 0, 1])):
+        # Transformation 3 (sta): aligns the joint's axis of movement with the z axis of the
+        # coordinate system. Only necessary for movable joints (therefore axis is not None) if
+        # the axis is not already parallel to the z axis
+        align_transformation = align_vectors(np.array([0, 0, 1]), axis)
+        align_quat = ScipyRotation.from_matrix(align_transformation).as_quat()
+        sta = [name + '_sta',
+               {'qw': align_quat[3],
+                'qx': align_quat[0],
+                'qy': align_quat[1],
+                'qz': align_quat[2]},
+               [],
+               ]
+
+        # Transformation 5 (unsta): after aligning with the z axis and adding motion of movable
+        # joint, reverses the alignment so that we are still in the correct coordinate system.
+        # (Note that the movement of the joint is applied between sta and unsta)
+        unalign_transformation = np.linalg.inv(align_transformation)
+        unalign_quat = ScipyRotation.from_matrix(unalign_transformation).as_quat()
+        unsta = [name + '_unsta',
+                 {'qw': unalign_quat[3],
+                  'qx': unalign_quat[0],
+                  'qy': unalign_quat[1],
+                  'qz': unalign_quat[2]},
+                 [],
+                 ]
+
+        joint_transformations.append(sta)
+        # unsta will be appended after the joint movement transformation
+
+    # Transformation 4 (mov): represents the movement of the joint. This is the only transformation
+    # with state variables.
+    if type_ in ['continuous', 'revolute']:
+        mov = [name + '_mov', {'rz': 0}, ['rz']]
+    elif type_ == 'prismatic':
+        mov = [name + '_mov', {'tz': 0}, ['tz']]
+    # Commenting out floating and planar joints for now because we cannot test them by comparing

```

Comment:
Delete unused code snippets. If you want to add them later, open an issue where you describe how to add them and what tests need to be performed.

Also mind that while other libraries dont support the joint it is possible to construct toy examples with analytic solutions which can be used as tests